File: core/src/com/gdxjam/magellan/GameObj.java
package com.gdxjam.magellan;

import aurelienribon.tweenengine.TweenManager;
import com.gdxjam.magellan.screen.BaseScreen;

/**
 * Created by lolcorner on 19.12.2015.
 */
public class GameObj {

    public Sector sector;
    public enum Factions{NEUTRAL, PLAYER, ENEMY, PIRATE};
    public Factions faction = Factions.NEUTRAL;
    public String submenuOpen = "";

    public GameObj(Sector sector){
        this.sector = sector;
        sector.gameObjs.add(this);
    }
    public void tick(){

    }
    public void render(float deltaTime) {

    }

    public void closeWindow() {
        MagellanGame.instance.windowScreen.closeWindow();
    }

    public void showInteractionWindow() {
        closeWindow();
        MagellanGame.instance.windowScreen.showInteractionWindow((IDrawableWindow) this);
    }
}

File: core/src/com/gdxjam/magellan/gameobj/GameObj.kt
package com.gdxjam.magellan.gameobj

import com.gdxjam.magellan.MagellanGame
import com.gdxjam.magellan.models.Sector

/**
 * Created by lolcorner on 19.12.2015.
 */
abstract class GameObj(@JvmField var sector: Sector) {

    @JvmField
    var faction = Factions.NEUTRAL
    @JvmField
    var submenuOpen = ""

    enum class Factions {
        NEUTRAL, PLAYER, SAATOO, PIRATE
    }

    init {
        sector.gameObjs.add(this)
    }

    open fun passiveTick() {}
    open fun activeTick() {}
    open fun render(deltaTime: Float) {}

    fun closeWindow() {
        MagellanGame.instance.windowScreen.closeWindow()
    }

    fun showInteractionWindow() {
        closeWindow()
        MagellanGame.instance.windowScreen.showInteractionWindow(this as IDrawableWindow)
    }
}
